compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/InterfaceLowering.kt

```diff
@@ -243,7 +250,31 @@ internal class InterfaceLowering(val context: JvmBackendContext) : IrElementTran
         val newFunction = removedFunctions[expression.symbol]?.owner
         return super.visitCall(
             if (newFunction != null) {
-                irCall(expression, newFunction, receiversAsArguments = true)
+                IrCallImpl(
+                    expression.startOffset,
+                    expression.endOffset,
+                    expression.type, // TODO: Here we copy the old type, but really it should be [newFunction.returnType] under an appropriate substitution
```

Comment:
Apart from potential references to the type parameters of the interface instead of the default method, wouldn't these be the same type?

Reply:
You are right! And the frontend would have done the substitution as we saw in #3009, I guess :)